src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.BulldozerProduser;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.ExcavatorProduser;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.TruckProduser;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private BulldozerProduser bulldozerProduser = new BulldozerProduser();
```

Comment:
```suggestion
    private MachineProducer bulldozerProduser = new BulldozerProduser();
```
хорошою практикою вважається використання більш абстрактного типу для посилання